app/vmalert/notifier/template.go

```diff
@@ -31,6 +33,67 @@ import (
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promutils"
 )
 
+// go template execution fails when it's tree is empty
+const defaultTemplate = `{{- define "default.template" -}}{{- end -}}`
+
+type textTemplate struct {
+	current     *textTpl.Template
+	replacement *textTpl.Template
+}
+
+var masterTmpl textTemplate
```

Comment:
For me, it looks like data race is possible with masterTmpl.

 When alert executed and configuration file reloaded at the same time.

Is it possible to protect with RWMutex and make it thread safe?

 I suggest to introduce methods - `Current()` and `CurrentClone` with readlocks
and add writelocks to the `LoadTemplates` and `ReplaceTemplates`

 I suggest to protect `textTemplate` with RWMutex.